backend/baseclass.pm

```diff
@@ -168,145 +166,133 @@ sub _write_buffered_data_to_file_handle {
     }
 }
 
-=head2 run_capture_loop($timeout)
-
-=out
-
-=item timeout
-
-run the loop this long in seconds, indefinitely if undef, or until the
-$self->{cmdpipe} is closed, whichever occurs first.
+sub do_capture {
+    my ($timeout, $starttime, $buckets, $wait_time_limit, $hits_limit) = @_;
+    last unless $self->{cmdpipe};
```

Comment:
`last` outside a loop?

Reply:
ok, that's going too far :) I removed that commit from the PR for now.